app/org/sagebionetworks/bridge/upload/TranscribeConsentHandler.java

```diff
@@ -39,9 +45,15 @@ public void handle(@Nonnull UploadValidationContext context) {
 
         // Also get external ID
         String userExternalId = options.get(ParticipantOption.EXTERNAL_IDENTIFIER);
+        
+        // And get user data groups
+        String dataGroupsSer = options.get(ParticipantOption.DATA_GROUPS);
+        Set<String> userDataGroups = (isNotBlank(dataGroupsSer))
+                ? Sets.newHashSet(Splitter.on(",").split(dataGroupsSer)) : null;
```

Comment:
We had a bug where the externalId wasn't being propagated to the record because of a bug in the copy-constructor. To make sure we don't have a similar bug here, I highly recommend also addeding data groups to https://github.com/Sage-Bionetworks/BridgePF/blob/develop/test/org/sagebionetworks/bridge/upload/TranscribeConsentHandlerTest.java

This is admittedly non-obvious and won't scale, so long-term, we'll need to figure out a better solution for this.